packages/orchestration/src/utils/progress.js

```diff
@@ -0,0 +1,178 @@
+/**
+ * @file Utilities for working with ResultMeta<T>.
+ */
+
+import { isVow } from '@agoric/vow/src/vow-utils.js';
+
+/**
+ * @import {EVow, VowTools, Fulfilled} from '@agoric/vow';
+ * @import {Zone} from '@agoric/base-zone';
+ */
+
+/**
+ * @typedef {Record<string, any>} ProgressReport
+ */
+
+/**
+ * @template T
+ * @typedef {object} ReporterPublication
+ * @property {T} value
+ * @property {EVow<ReporterPublication<T>>} tail
+ */
+
+/**
+ * @template T
+ * @template [A=unknown]
+ * @callback PublicationReducer Called on the value of each linked T. This
+ *   function should perform any side-effects it can when it receives a new
+ *   value, since there is no guarantee that the ProgressRecord tail will be
+ *   prompt (it may take a long time with cross-chain communication) or ever
+ *   fulfill (it may result in a rejection).
+ * @param {T | undefined | null} value `value == null` is a sentinel for the
+ *   final reducer call.
+ * @param {A} accum previous reduction value
+ * @returns {EVow<A | null>} Returning `null` exits the reduction
+ */
+
+/**
+ * Utilities for working with ResultMeta within an async flow.
+ */
+export const progressTrackerAsyncFlowUtils = {
+  /**
+   * Converts a specimen to a Promise. Since the use of Vows is only needed
+   * outside of async-flows, just reject if we encounter one.
+   *
+   * @template T
+   * @param {EVow<T>} specimen
```

Comment:
since this fails when the specimen is a vow, why take `EVow` as the param type?

```suggestion
   * @param {ERef<T>} specimen
```

Reply:
This was an overly-narrow type.  `specimen` is best typed as `unknown` or, in this case, `T`, so I've done that now.